### DISCUSSION TITLE
WIX0200: The Component element contains an unhandled extension element 'XmlConfig'. Please ensure that the extension for elements in the 'http://wixtoolset.org/schemas/v4/wxs/util' namespace has been provided.

### DISCUSSION THREAD

**Discussion by DipakTated** [wixtoolset/issues#7886] [Questions]
Why do i see this error

error WIX0200: The Component element contains an unhandled extension element 'XmlConfig'. Please ensure that the extension for elements in the 'http://wixtoolset.org/schemas/v4/wxs/util' namespace has been provided. 

I have wix 4 installed 4.0.2+644ed011
 and wix toolset of 3.14 at my folder  C:\Program Files (x86)\WiX Toolset v3.14

Code is as follows :

<Wix xmlns="http://wixtoolset.org/schemas/v4/wxs"
     xmlns:util="http://wixtoolset.org/schemas/v4/wxs/util"
     xmlns:xs="http://www.w3.org/2001/XMLSchema-instance">
  <?include Variables.wxi ?>
  <Fragment>
    <DirectoryRef Id="INSTALLFOLDER">
      <Component Id="CommonConfigSettings">
        <util:XmlConfig Id="RemoteApiUrlValue" File="$(var.ConfigPath)" Action="create" Node="value" On="install" ElementPath="//configuration/appSettings/add[\[]@key='RemoteApiUrl'[\]]" Name="value" Value="[REMOTEAPIURL]" />
        <util:XmlConfig Id="LanguageValue" File="$(var.ConfigPath)" Action="create" Node="value" On="install" ElementPath="//configuration/appSettings/add[\[]@key='Language'[\]]" Name="value" Value="[LANGUAGE]" />
        <util:XmlConfig Id="UseMachineProtectionValue" File="$(var.ConfigPath)" Action="create" Node="value" On="install" ElementPath="//configuration/configProtectedData/providers/add[\[]@name='DataProtectionConfigurationProvider'[\]]" Name="useMachineProtection" Value="[USEMACHINEPROTECTION]" />
      </Component>
    </DirectoryRef>
  </Fragment>
  <Fragment>
    <DirectoryRef Id="INSTALLFOLDER">
      <Component Id="CommonRegistryEntries">
        <RegistryValue Root="HKMU" Key="Software\[PRODUCTTITLE]" Name="RemoteApiUrl" Value="[REMOTEAPIURL]" Type="string" />
        <RegistryValue Root="HKMU" Key="Software\[PRODUCTTITLE]" Name="Installed" Value="1" Type="integer" KeyPath="yes" />
      </Component>
    </DirectoryRef>
  </Fragment>
  <Fragment>
    <DirectoryRef Id="INSTALLFOLDER">
      <Component Id="RegistryAppUri">
        <RegistryKey Root="HKCR" Key="[AppUriSchema]" ForceCreateOnInstall="yes" ForceDeleteOnUninstall="yes">
          <RegistryValue Type="string" Value="URL:[AppUriSchema]" KeyPath="yes" />
          <RegistryValue Type="string" Name="URL Protocol" Value="" />
        </RegistryKey>
        <RegistryKey Root="HKCR" Key="[AppUriSchema]\shell\open\command" ForceCreateOnInstall="yes" ForceDeleteOnUninstall="yes">
          <RegistryValue Type="string" Value="&quot;[INSTALLFOLDER][APPNAME]&quot; [AppUriSchemaParams]" />
        </RegistryKey>
      </Component>
    </DirectoryRef>
  </Fragment>
  <Fragment>
    <ComponentGroup Id="CommonComponents" Directory="INSTALLFOLDER">
      <ComponentRef Id="CommonConfigSettings" />
      <ComponentRef Id="CommonRegistryEntries" />
      <ComponentRef Id="RegistryAppUri" />
    </ComponentGroup>
  </Fragment>
</Wix>

**Comment by DipakTated** [comment]
[issuewix.txt](https://github.com/wixtoolset/issues/files/13378606/issuewix.txt)
<Wix xmlns="http://wixtoolset.org/schemas/v4/wxs"
     xmlns:util="http://wixtoolset.org/schemas/v4/wxs/util"
     xmlns:xs="http://www.w3.org/2001/XMLSchema-instance">
  <?include Variables.wxi ?>
  <Fragment>
    <DirectoryRef Id="INSTALLFOLDER">
      <Component Id="CommonConfigSettings">
        <util:XmlConfig Id="RemoteApiUrlValue" File="$(var.ConfigPath)" Action="create" Node="value" On="install" ElementPath="//configuration/appSettings/add[\[]@key='RemoteApiUrl'[\]]" Name="value" Value="[REMOTEAPIURL]" />
        <util:XmlConfig Id="LanguageValue" File="$(var.ConfigPath)" Action="create" Node="value" On="install" ElementPath="//configuration/appSettings/add[\[]@key='Language'[\]]" Name="value" Value="[LANGUAGE]" />
        <util:XmlConfig Id="UseMachineProtectionValue" File="$(var.ConfigPath)" Action="create" Node="value" On="install" ElementPath="//configuration/configProtectedData/providers/add[\[]@name='DataProtectionConfigurationProvider'[\]]" Name="useMachineProtection" Value="[USEMACHINEPROTECTION]" />
      </Component>
    </DirectoryRef>
  </Fragment>
  <Fragment>
    <DirectoryRef Id="INSTALLFOLDER">
      
      
      <Component Id="CommonRegistryEntries">
        <RegistryValue Root="HKMU" Key="Software\[PRODUCTTITLE]" Name="RemoteApiUrl" Value="[REMOTEAPIURL]" Type="string" />
        <RegistryValue Root="HKMU" Key="Software\[PRODUCTTITLE]" Name="Installed" Value="1" Type="integer" KeyPath="yes" />
      </Component>
    </DirectoryRef>
  </Fragment>
  <Fragment>
    <DirectoryRef Id="INSTALLFOLDER">
      <Component Id="RegistryAppUri">
        <RegistryKey Root="HKCR" Key="[AppUriSchema]" ForceCreateOnInstall="yes" ForceDeleteOnUninstall="yes">
          <RegistryValue Type="string" Value="URL:[AppUriSchema]" KeyPath="yes" />
          <RegistryValue Type="string" Name="URL Protocol" Value="" />
        </RegistryKey>
        <RegistryKey Root="HKCR" Key="[AppUriSchema]\shell\open\command" ForceCreateOnInstall="yes" ForceDeleteOnUninstall="yes">
          <RegistryValue Type="string" Value="&quot;[INSTALLFOLDER][APPNAME]&quot; [AppUriSchemaParams]" />
        </RegistryKey>
      </Component>
    </DirectoryRef>
  </Fragment>
  <Fragment>
    <ComponentGroup Id="CommonComponents" Directory="INSTALLFOLDER">
      <ComponentRef Id="CommonConfigSettings" />
      <ComponentRef Id="CommonRegistryEntries" />
      <ComponentRef Id="RegistryAppUri" />
    </ComponentGroup>
  </Fragment>
</Wix>

**Comment by NJullienSweet** [comment]
I have the same error for `bal`.
```xml
<?xml version="1.0" encoding="UTF-8"?>
<Wix xmlns="http://wixtoolset.org/schemas/v4/wxs" xmlns:bal="http://wixtoolset.org/schemas/v4/wxs/bal">
	<Bundle Version="0.0.0.0" UpgradeCode="09974739-b91b-4e1e-a942-bd8759319f6a">
		<BootstrapperApplication>
			<bal:WixStandardBootstrapperApplication LicenseFile="License.rtf" LogoFile="Logo.png" Theme="rtfLargeLicense" />
		</BootstrapperApplication>
		<Chain>
			<!-- [...] -->
		</Chain>
	</Bundle>
</Wix>
```
```
error WIX0200: The BootstrapperApplication element contains an unhandled extension element 'WixStandardBootstrapperApplication'. Please ensure that the extension for elements in the 'http://wixtoolset.org/schemas/v4/wxs/bal' namespace has been provided.
```

**Comment by barnson** [comment]
Please provide a [minimal reproducible example](https://en.wikipedia.org/wiki/Minimal_reproducible_example) as an attachment or GitHub repo.

**Comment by jcmarcanotrias** [comment]
Same problem here:

```
<Wix xmlns="http://wixtoolset.org/schemas/v4/wxs" xmlns:util="http://wixtoolset.org/schemas/v4/wxs/util">
	<Package Name="PAppScriptsInstaller" Manufacturer="xxxxxx" Version="1.0.0.0" UpgradeCode="9a2b6d7e-60b6-4c2b-9f52-523c41dd55ad">

		<MajorUpgrade DowngradeErrorMessage="!(loc.DowngradeError)" />

		<MediaTemplate EmbedCab="yes"/>

		<Launch Condition="MsiNTProductType = 1" Message="Not valid system"/>

		<util:QueryNativeMachine />

		<Launch Condition="1" Message="WIX_NATIVE_MACHINE"/>
		
		<Feature Id="Main">
			<ComponentGroupRef Id="ConsoleApp1" />
		</Feature>
	</Package>
</Wix>

```

WIX0200	The Package element contains an unhandled extension element 'QueryNativeMachine'. Please ensure that the extension for elements in the 'http://wixtoolset.org/schemas/v4/wxs/util' namespace has been provided.

**Reply by jcmarcanotrias** [reply]
I solved it updating from version 4.0.4 to 5.0.1. 

Version 5.0.0 also works.

versions 4.x.x does not work

**Reply by jcmarcanotrias** [reply]
To show the environment variable value, I changed the code


		<util:QueryNativeMachine />

		<Launch Condition="TRUE" Message="[WIX_NATIVE_MACHINE]"/>

**Comment by bevanweiss** [comment]
I suspect the cause of these error messages is that the wixext is not referenced in the project(s)
Like C# libraries, if you use an extension, you need to bring it into the project.  Nuget is the answer..
![image](https://github.com/user-attachments/assets/14f1b43f-286a-4df4-a38d-4585cbf1ee06)

For each non-wxs namespace, you're probably going to need an extension (.wixext) referenced by the project.

**Comment by rohan-raut** [comment]
I am getting the same error for elements **PermissionEx** and **FirewallException** for version 5.0.2.
 
```
error WIX0200: The File element contains an unhandled extension element 'FirewallException'. Please ensure that the extension for elements in the '[http://wixtoolset.org/schemas/v4/wxs/firewall'](http://wixtoolset.org/schemas/v4/wxs/firewall%27) namespace has been provided.
```
 
```
error WIX0200: The CreateFolder element contains an unhandled extension element 'PermissionEx'. Please ensure that the extension for elements in the '[http://wixtoolset.org/schemas/v4/wxs/util'](http://wixtoolset.org/schemas/v4/wxs/util%27) namespace has been provided.
```
 
I have also provided the namespaces as below:
```
<Wix xmlns="http://wixtoolset.org/schemas/v4/wxs" 
   xmlns:util="http://wixtoolset.org/schemas/v4/wxs/util" 
   xmlns:fire="http://wixtoolset.org/schemas/v4/wxs/firewall">
..........
```

**Reply by bevanweiss** [reply]
What does your authoring look like for the FirewallException and PermissionEx (i.e. include as code here, or link to repository).
They should look like `<fire:FirewallException ....>` and `<util:PermissionEx ....>`

**Reply by rohan-raut** [reply]
Please refer the below code,
```
<File Id="..." Name="..." Source="..." KeyPath="no">
    <fire:FirewallException Id="..." Name="..." IgnoreFailure="yes" Scope="any" />
</File>
```
```
<CreateFolder>
    <util:PermissionEx User="Everyone" GenericAll="yes" />
</CreateFolder>
```

And my .wixproj file has the following PackageReferences:
```
<ItemGroup>
    <PackageReference Include="WixToolset.Heat" Version="5.0.2" />
    <PackageReference Include="WixToolset.Util.wixext" Version="5.0.2" />
    <PackageReference Include="WixToolset.UI.wixext" Version="5.0.2" />
    <PackageReference Include="WixToolset.NetFx.wixext" Version="5.0.2" />
    <PackageReference Include="WixToolset.Firewall.wixext" Version="5.0.2" />
</ItemGroup>
